pkg/apis/apps/validation/validation_test.go

```diff
@@ -42,6 +42,38 @@ func intStrAddr(intOrStr intstr.IntOrString) *intstr.IntOrString {
 	return &intOrStr
 }
 
+type statefulSetTweak func(ss apps.StatefulSet)
+
+func mkStatefulSet(template *api.PodTemplate, tweaks ...statefulSetTweak) apps.StatefulSet {
+	s := apps.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+		Spec: apps.StatefulSetSpec{
+			PodManagementPolicy: apps.OrderedReadyPodManagement,
+			Selector:            &metav1.LabelSelector{MatchLabels: map[string]string{"a": "b"}},
+			Template:            template.Template,
+			UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
+		},
+	}
+
+	for _, tw := range tweaks {
+		tw(s)
```

Comment:
This needs to be a pointer - I see you did that in the last commit, but each of these commits, as beautiful as they are, don't work?  Can you back-populate that into these commits?